src/mono-runtimes/mono-runtimes.targets

```diff
@@ -280,26 +278,37 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_BuildRuntimes"
-      Condition=" '@(_MonoRuntime)' != '' Or '@(_MonoCrossRuntime)' != '' "
-      DependsOnTargets="_GetRuntimesOutputItems"
-      Outputs="@(_RuntimeSource);@(_RuntimeBinarySource);@(_CrossRuntimeBinarySource);@(_ProfilerSource);@(_MonoPosixHelperSource);@(_RuntimeEglibHeaderSource);@(_MonoBtlsSource);@(_BclTestOutput)">
+  <Target Name="_DownloadArchive"
+      DependsOnTargets="_GetMonoGitCommitHash" >
+    <MakeDir
+        Directories="$(MonoSourceFullPath)\sdks\out"
+    />
     <Exec
-        Command="make DISABLE_IOS=1 $(MakeConcurrency) @(_MonoRuntime->'package-android-%(Identity)', ' ') @(_MonoCrossRuntime->'package-android-%(Identity)', ' ') $(_MonoSdksParameters)"
+        Condition=" !Exists('$(MonoSourceFullPath)\sdks\out\.stamp-android-$(_MonoSdksConfiguration)-$(_MonoGitCommitHash)-$(HostOS)-download') "
```

Comment:
A problem with this condition is that it's plausibly "tying" this command to mono internals which could change in the future, specifically the filename of the `.stamp-...` file that mono will produce.

Would it be saner to instead update the make targets to accept a make variable containing the `.stamp` file to produce?  The "fallback source build path" would thus become:

```
make DISABLE_IOS=1 STAMP_SUCCESS_PATH=".stamp-from-xamarin-android" $(MakeConcurrency) @(_MonoRuntime->'package-android-%(Identity)', ' ') @(_MonoCrossRuntime->'package-android-%(Identity)', ' ') @(_MonoBcl->'package-android-%(Identity)', ' ') @(_LlvmRuntime->'provision-llvm-%(Identity)', ' ') $(_MonoSdksParameters)
```

Reply:
The `.stamp-android-$(_MonoSdksConfiguration)-$(_MonoGitCommitHash)-$(HostOS)-download` file is created at https://github.com/xamarin/xamarin-android/pull/2265/files#diff-50d8c6c6db025c3af45ee3e9669eee01R294 so not internal to Mono.